inference-explainers/inference_api.md

```diff
@@ -0,0 +1,263 @@
+# Inference API
+
+**Authors:** <br>
+[Steve Gan][5], Google Privacy Sandbox <br>
+
+## Introduction
+
+This document provides a comprehensive overview of the Bidding and Auction (B&A) Inference service API with externally supplied models ([B&A inference overview][1]), including supported machine learning (ML) frameworks, model formats, model management processes, and the JavaScript API for executing inference tasks.
+
+## Supported Machine Learning Frameworks and Model Formats
+
+The Tensorflow and PyTorch runtimes are supported, each with a dedicated inference service sidecar binary. Currently, we offer the Tensorflow v2.14.0 sidecar binary and the PyTorch v2.1.1 sidecar binary.
+
+For Tensorflow, the SavedModel format is used while PyTorch uses the TorchScript format. We recommend that ad techs develop and train models in an environment in which the ML runtime version aligns with the intended sidecar.
+
+The maximum size for a single ML model is 2GB.
+
+## Model management
+
+ML models used by the inference service are fetched periodically from a linked cloud storage (e.g., [Google Cloud Storage buckets][2], [Amazon S3 buckets][3]) and made available for serving. Each model is uniquely identified by its path within the cloud storage. To determine the models to load, the inference service looks for a JSON model configuration file in the cloud bucket. Ad techs should maintain a model configuration file in JSON format within the same cloud storage as the models. The B&A service periodically checks the configuration file (according to period configurable by ad techs) for any changes and triggers the loading of new models into the inference service sidecar’s memory as needed.
```

Comment:
one small suggestion: can we also mention how the system knows that the JSON model config file is? Maybe mention the config setting for this ...